### ISSUE TITLE
isobaric substitutions in PepQuery

### ISSUE THREAD

**Issue by Seb-Leb** [bzhanglab/PepQuery#7]
This is a question about how PepQuery handles isobaric substitutions. We search two different spectra files for the same peptide with the same protein reference database. PepQuery indicates that a spectrum from on of the files fits better to our peptide than any proteins in the reference DB with unrestricted modification search. However in the other file, only a spectrum scored higher on a protein in the reference DB with a deamidation of N (isobaric to D) is assigned. 
We expected to see the PSM with the modified reference peptide in the first analysis as well but it doesn't show.

The Genome Reasearch Publication clearly describes how the scores are calculated but I have trouble understanding how the score between an isobaric PTM peptide and the queried peptide are different. Since the deamidation of N is the exact mass of D should both PSMs not be assigned the same score?

My mgf files with PepQuery results can be downloaded from here:
https://drive.google.com/file/d/145Ne9qpGQTu_fpFg0jXiS3-DVHXDpJm6/view?usp=sharing
(Fr4S31 is the one where our peptide is most confident and Fr2S31 is the one where a better matched N[deamidated]/D reference protein)

thank you!

**Comment by wenbostar** [comment]
Could you send me your command line?

**Comment by wenbostar** [comment]
By the way, it looks your data is not centroid as shown below:
![image](https://user-images.githubusercontent.com/5302110/62589161-f9a8ce80-b88d-11e9-8f54-bbb1acbd154c.png)

**Comment by Seb-Leb** [comment]
Here is the command I run:

java -jar pepquery.jar -um -cpu 2 -pep Fr2S31.txt -db refprots.fasta -ms Fr2S31.mgf -o pepq_isobar_test/Fr2S31/

Do I have to pre-process the data? (we just converted it from raw with msconvert..)

**Comment by wenbostar** [comment]
What's the command line you used to convert raw to mgf?

**Comment by Seb-Leb** [comment]
My colleague did it using the GUI
is it because of peak picking?

**Comment by wenbostar** [comment]
From raw MS/MS data to MGF data:

msconvert --filter "peakPicking true 1-2" --mgf *.raw

From mzML format data to MGF data:

msconvert --filter "peakPicking true 1-2" --mgf *.mzML

**Comment by Seb-Leb** [comment]
Ok thanks. I will try that.

**Comment by wenbostar** [comment]
> My colleague did it using the GUI
> is it because of peak picking?

Yes. If you use msconvert GUI, make sure you select peak picking option:
![image](https://user-images.githubusercontent.com/5302110/62634481-95245880-b8fb-11e9-9a87-e154d5e3b36c.png)

**Comment by Seb-Leb** [comment]
This seems to have solved the issue.
thank you

**Comment by Seb-Leb** [comment]
Hello, I am sorry to re-open this issue but it seems that we observed the same behavior with files that are centroided. I added our reference dataset as well as two mgf files (H203.mgf if the file where PepQuery assings the best PSM to our peptide and Fr2S31.mgf is the file where PepQuery assigns a better (or equal) association score to a spectrum with an isobaric substitution on a reference protein).

[gdrive share link](https://drive.google.com/file/d/15JzcMPEKxJGYx9UBWg1d3dKjo-aJxqiq/view?usp=sharing)

The command used is as mentioned before.

Thank you

**Comment by wenbostar** [comment]
It looks you used the default setting for most of the parameters. What type of MS machine was used to generate your data?

**Comment by Seb-Leb** [comment]
QExactive

**Comment by wenbostar** [comment]
Then you should set -itol 0.05 not the default value 0.6.

**Comment by wenbostar** [comment]
In addition, you can add parameter [-hc](http://pepquery.org/document.html#saparameter) with more stringent filtering.

**Comment by Seb-Leb** [comment]
How should we use the -hc falg with PepQuery 1.0? It dosen't appear in the options when calling pepquery.jar help

2019-08-23 13:55:26 [ INFO] Start analysis
2019-08-23 13:55:26 [DEBUG] -h
java -Xmx2G -jar pepquery.jar
Version 1.0.0
usage: Options
 -anno <arg>             Annotation files folder for VCF/BED/GTF
 -c <arg>                The max missed cleavages, default is 2
 -cpu <arg>              The number of cpus used, default is 1
 -db <arg>               Fasta format database file
 -e <arg>                1:Trypsin (default), 2:Trypsin (no P rule),
                         3:Arg-C, 4:Arg-C (no P rule), 5:Arg-N, 6:Glu-C,
                         7:Lys-C
 -f <arg>                The frame to translate DNA sequence to protein.
                         The right format is like this:
                         "1,2,3,4,5,6","1,2,3","1". "0" means to keep the
                         longest frame. In default, for each frame only
                         the longest protein is used.
 -fixMod <arg>           Fixed modification, the format is like : 1,2,3.
                         Default is 6 (Carbamidomethylation(C)[57.02])
 -fragmentMethod <arg>   1: CID/HCD (default), 2: ETD
 -h                      Help
 -i <arg>                Take protein, DNA or VCF as input
 -itol <arg>             Fragment ion m/z tolerance, default is 0.6da
 -m <arg>                Scoring method: 1=HyperScore (default), 2=MVH
 -maxCharge <arg>        The maximum charge to consider if the charge
                         state is not available, default is 3
 -maxVar <arg>           Max number of variable modifications, default is
                         3
 -minCharge <arg>        The minimum charge to consider if the charge
                         state is not available, default is 2
 -minPeaks <arg>         Min peaks in spectrum, default is 10
 -minScore <arg>         Minimum score to consider for peptide searching,
                         default is 5
 -ms <arg>               Spectrum file used for identification, mgf format
 -n <arg>                The number of random peptides, default is 1000
 -o <arg>                Output dir
 -pep <arg>              Peptide sequence which you want to search
 -prefix <arg>           Output file prefix
 -printPTM               Print PTMs
 -t <arg>                Input type:
                         1=>protein,2=>DNA,3=>VCF,4=>BED,5=>GTF
 -tol <arg>              Precursor ion m/z tolerance, default is 10
 -tolu <arg>             The unit of precursor ion m/z tolerance, default
                         is ppm
 -um                     Validation with unrestricted modification
                         searching
 -varMod <arg>           Variable modification, the format is like :
                         1,2,3. Default is 107 (Oxidation(M)[15.99])

**Comment by wenbostar** [comment]
Please try the [new version](http://pepquery.org/download.html) which has this option.
